### DISCUSSION TITLE
determine boundary plane (2D EBSD and FIB cross sections)

### DISCUSSION THREAD

**Discussion by Saoussen-ouhiba** [mtex-toolbox/mtex#968] [Ask Anything] [answered]
Hello, 

I would like to determine the boundary plane between grain (id =7835) and grain (id =7444) (shown in the figure below). For this purpose, two cross sections (CS1 and CS2) were performed along the two boundaries separating these two grains as seen in the figure below. I was able to determine the inclination of each boundary plane. Could you help me to calculate the trace of each boundary plane please? and how can I use these information to determine each boundary plane?

Thank you a lot for your help,

Saoussen 

![question](https://user-images.githubusercontent.com/73701277/115023783-87d8b300-9ebf-11eb-979c-998ddc20ccf3.JPG)

**Comment by kilir** [comment, marked as answer]
Hi,
you can do something like this :

```
inc = 97*degree+pi/2; % inclination, assuming z out of plane and that the left part of your fib section is is on the top of your map
azi = 100*degree; % trace of your fib section (me guessing, based on x-east / y-south)
gb_in_fibtrace = vector3d.byPolar(inc,azi)

% locate the grain boudanry in question in your EBSD map
% e.g.
% xy = ginput;
gb = grains.boundary(inpolygon(grains.boundary,[xy]))

grains_boudnary_normal_3d = cross(gb.direction,gb_in_fibtrace)
```
Cheers,
Rüdiger

**Reply by Saoussen-ouhiba** [reply]
Hello Rüdiger,

Thank you a lot. I have a question regarding the calculation of azi. Is it determined according to the figure below?

![question2](https://user-images.githubusercontent.com/73701277/115254087-4fd6a780-a12d-11eb-9699-b57ae048fdd7.JPG)


Otherwise, the script seems to be working well. I got the following result:

```
grains_boundary_normal_3d = vector3d (show methods, plot)
 size: 14 x 1
 antipodal: true
         x         y         z
         0  0.992546 -0.120018
         0 -0.992546  0.120018
         0  0.992546 -0.120018
         0  0.992546 -0.120018
         0 -0.992546  0.120018
  0.992546         0 0.0211624
         0  0.992546 -0.120018
         0 -0.992546  0.120018
         0  0.992546 -0.120018
         0  0.992546 -0.120018
         0  0.992546 -0.120018
  0.992546         0 0.0211624
         0 -0.992546  0.120018
         0 -0.992546  0.120018
```
Is it possible to get a kind of average value for the grains_boundary_normal_3d please? and how to estimate the deviation of this normal from (111) plane normal in degree?

Thank you a lot for your help,

Saoussen

**Reply by kilir** [reply]
Hi,
1) one way to get the trace of your fib section would be to trace use the two coordinates from its beginning and end, and assuming your have x-east, y-south in your map, the angle should be the one you indicated.

2) As far as I can see you also need to smooth your grains before getting the grain boundary direction. Once done that, you can for example use the mean(), either on the grain boundary normal or on the direction before computing the cross product.

3) To get the deviation from (111), you can compute the direction of the normal to (111) in specimen coordinates (orientation*Miller) and inspect the angle between both. In the case of {111} it might be easier to translate your result of the grain boundary normal to crystal coordinates:
```
gb_plane_crystal_coords = inv(grain(xy).meanOrientation).*grains_boundary_normal_3d
angle(gb_plane_crystal_coords,Miller(1,1,1,cs))
```
Cheers,
Rüdiger

**Reply by Saoussen-ouhiba** [reply]
Thank you a lot, Rüdiger. It is very clear to me now.
Have a nice day,

Saoussen
